hibernate/cache-commons/src/main/java/org/infinispan/hibernate/cache/commons/access/UnorderedDistributionInterceptor.java

```diff
@@ -80,9 +80,9 @@ private Object handleDataWriteCommand(InvocationContext ctx, DataWriteCommand co
 			List<Address> owners = cacheTopology.getDistribution(command.getKey()).writeOwners();
 			if (owners.contains(rpcManager.getAddress())) {
             return invokeNextAndHandle( ctx, command, (rCtx, rCommand, rv, throwable) -> {
-               Object remoteInvocation = invokeRemotelyAsync(owners, rCtx, (WriteCommand) rCommand);
+               CompletionStage<?> remoteInvocation = invokeRemotelyAsync(owners, rCtx, (WriteCommand) rCommand);
                if (remoteInvocation != null) {
-                  return ((CompletionStage<?>) remoteInvocation).handle((responses, t) -> rv);
+                  return remoteInvocation.thenApply(responses -> rv);
```

Comment:
doesn't this imply that futures chained to the returned stage will not be executed whereas before with handle they would always run?

Reply:
Handling of exceptions has already been fixed in the latter commit for ISPN-10267.